### ISSUE TITLE
Bug: exception using 'updateEvent' and changing allDay from false to true

### ISSUE THREAD

**Issue by marcoancona** [fullcalendar/fullcalendar#3518]
An exception occurs when a non-daily event is updated to be a daily event. Please see [this fiddle](http://jsfiddle.net/anconam/jRFYE/856/) (the event is automatically updated after 2 seconds).

```
VM2146 fullcalendar.js:650 Uncaught TypeError: Cannot read property 'clone' of null
```

The problems seems to arise in the `mutateEvent()` method:

```js
...
normalizeEventDates(oldProps);  // <-- this will nullify oldProps.end because end == start (we strip the time so they will both be midnight of the original date)

// need to clear the end date if explicitly changed to null
clearEnd = event._end !== null && newProps.end === null;

// compute the delta for moving the start date
startDelta = diffDates(newProps.start, oldProps.start);

// compute the delta for moving the end date
if (newProps.end) {
	endDelta = diffDates(newProps.end, oldProps.end);  // <-- since oldProps.end is null, this raise an exception
	durationDelta = endDelta.subtract(startDelta);
}
...
```

**Comment by arshaw** [comment]
recreated, thanks

**Comment by westy92** [comment]
Any update on this?

**Comment by arshaw** [comment]
In a recent refactor, which is now pushed to `master`, this is fixed. test I made:

https://github.com/fullcalendar/fullcalendar/commit/0d69aca083f5d3e75ad3d47c2d94ea95be380400

will be release soon.

**Comment by dab1211** [comment]
Hi @arshaw ,
What changes need to be made in the fullcalendar.js file for this to be fixed??
I do not see a fullcalendar.js file in master :/
Thanks!!!

**Comment by espen** [comment]
@dab1211 npm install and gulp dist should give you the file based on master. See https://github.com/fullcalendar/fullcalendar/blob/master/CONTRIBUTING.md#getting-set-up

**Comment by dab1211** [comment]
Thanks @espen ,
I dont have a way of doing that gulp. Any idea how else I can do this online or where i can generate the JS file?

**Comment by arshaw** [comment]
Released in v3.5.0 ([blog post](https://fullcalendar.io/blog/2017/08/bootstrap-support-and-future-developments/#more-257))
